### PR TITLE
Fix CDash test output truncation for ATDM Trilinos builds

### PR DESCRIPTION
**Pull request by bartlettroscoe**
This pulls in the TriBITS changes shown at:

* https://github.com/TriBITSPub/TriBITS/compare/6165385bda2cb968e036530f9718bbcecbef704f..d953bd2620f1c94e22b96a977797758e45396c0a?expand=1

The addresses TriBITSPub/TriBITS#326  and https://github.com/TriBITSPub/TriBITS/commit/990ebeeb60be76433c64888cc508937c314f51ca and otherwise little other issues.  See the list of commits at:

* https://github.com/TriBITSPub/TriBITS/commits/d953bd2620f1c94e22b96a977797758e45396c0a

after the commit https://github.com/TriBITSPub/TriBITS/commit/6165385bda2cb968e036530f9718bbcecbef704f.

Merging this will give us full build stats for platforms like 'ats2' and 'cts1' as part of #7376 and #7508.

## How was this tested?

The TriBITS test suite was run and 100% passed (see evidence in commit https://github.com/TriBITSPub/TriBITS/commit/d953bd2620f1c94e22b96a977797758e45396c0a).

### CODE REVIEW

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake [hunk 1]
```diff
@@ -1004,31 +1004,46 @@ INCLUDE(${CMAKE_CURRENT_LIST_DIR}/TribitsCTestDriverCoreHelpers.cmake)
...
-# ``TRIBITS_CTEST_DRIVER()`` is called and can be overridden in the env.
+# ``TRIBITS_CTEST_DRIVER()`` is called and can be overridden in the env when
+# running the CTest -S driver script.
 #
 #   .. _CTEST_TEST_TYPE:
 #
 #   ``CTEST_TEST_TYPE=[Nightly|Continuous|Experimental]``
 #
-#     Determines the type of build.  This value is passed in as the first
+#     Determines the model for build.  This value is passed in as the first
```
Comment (jmgate) [inline cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake:1014]: "the model for build" sounds strange to me.

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake [hunk 2]
```diff
@@ -1004,31 +1004,46 @@ INCLUDE(${CMAKE_CURRENT_LIST_DIR}/TribitsCTestDriverCoreHelpers.cmake)
...
 #     argument to the built-in CTest function ``CTEST_START()``.  Valid values
-#     include ``Nightly``, ``Continuous``, and ``Experimental``.  This also
-#     defines the default value for `${PROJECT_NAME}_TESTING_TRACK`_ as well
-#     as defines the default value for
+#     include ``Nightly``, ``Continuous``, and ``Experimental``.  As far a
```
Comment (jmgate) [inline cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake:1016]:
```suggestion
#     include ``Nightly``, ``Continuous``, and ``Experimental``.  As far as
```

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake [hunk 3]
```diff
@@ -1004,31 +1004,46 @@ INCLUDE(${CMAKE_CURRENT_LIST_DIR}/TribitsCTestDriverCoreHelpers.cmake)
...
+#     CTest is concerned, the only real impact this CTest "Model" has is on
+#     setting the time stamp in the build stamp (which is stored in the file
+#     ``Testing/TAG``).  For the model ``Nightly``, the time stamp in the
+#     build stamp is taken from the variable ``CTEST_NIGHTLY_START_TIME`` read
+#     in from the file `<projectDir>/CTestConfig.cmake`_ file.  Otherwise, the
+#     time stamp used is the current build start time.  (The reason this is
+#     significant is that builds on CDash that have the same site, buildname,
+#     and buildstamp are considered the same build and will combine results.)
+#     This also defines the default value for `${PROJECT_NAME}_TRACK`_ (see
+#     below) as well as defines the default value for
```
Comment (jmgate) [inline cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake:1026]: Sometimes the `*stamp` words are two words (`build stamp`) and sometimes they're combined (`buildstamp`).

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake [hunk 4]
```diff
@@ -1004,31 +1004,46 @@ INCLUDE(${CMAKE_CURRENT_LIST_DIR}/TribitsCTestDriverCoreHelpers.cmake)
...
+#
+#     Specifies the testing track that specifies the CDash group for which
+#     results are displayed under (i.e. the "Group" filter field on CDash).
+#     This is the value used for the (deprecated) ``TRACK`` argument (renamed
+#     ``GROUP`` in CMake/CTest versions 3.16+) of the built-in CTest function
+#     ``CTEST_START()``.  The default value is set to ``${CTEST_TEST_TYPE}``.
+#     However, if ``CTEST_TEST_TYPE==Experimental`` (or ``EXPERIMENTAL``),
+#     then ``${PROJECT_NAME}_TRACK`` is forced to ``Experimental``, even if it
+#     was set to a different value.  The default value can also be set in the
+#     ctest -S driver script itself by setting by setting
```
Comment (jmgate) [inline cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake:1042]:
```suggestion
#     ctest -S driver script itself by setting
```

File: cmake/tribits/ctest_driver/TribitsCTestDriverCoreHelpers.cmake [hunk 5]
```diff
@@ -1368,10 +1372,6 @@ MACRO(TRIBITS_CTEST_ALL_AT_ONCE)
     ELSE()
       MESSAGE("Configure PASSED!")
       SET(AAO_CONFIGURE_PASSED TRUE)
-      # Load target properties and test keywords
-      CTEST_READ_CUSTOM_FILES(BUILD "${CTEST_BINARY_DIRECTORY}")
-      # Overridde from this file!
-      INCLUDE("${TRIBITS_PROJECT_ROOT}/CTestConfig.cmake")
```
Comment (jmgate) [inline cmake/tribits/ctest_driver/TribitsCTestDriverCoreHelpers.cmake:1374]: Just making sure these lines were supposed to be removed.  I couldn't figure out a reason just looking at them.

File: cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst [hunk 6]
```diff
@@ -6304,6 +6304,41 @@ The following steps describe how to submit results to a CDash site using the
...
+  <special_group>)`` in the CTest -S ``*.cmake`` driver script itself or can
+  set it in the environment when running the ctest -S driver script.  For
+  example::
+
+    $ env <Project>_TRACK=<special_group> ... \
+      ctest -V -S <ctest_driver>.cmake
+
+  If the "build type" for the CDash group ``<special_group>`` was set to
+  "Daily", then set `CTEST_TEST_TYPE`_ to ``Nightly``.  Otherwise,
+  ``CTEST_TEST_TYPE`` can be set to ``Continuous`` or ``Experimental``.
```
Comment (jmgate) [inline cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst:6339]: I don't think this is valid reStructuredText for enumerated lists.  I think you'll need
```
1) Create the new CDash group ``<special_group>`` for CDash project
   ``<ProjectName>`` on the targeted CDash site.

   If the CDash group ``<special_group>`` is not already created, then one can
   create it by first logging into CDash with an account that can modify the
   etc.
```

for it to get parsed correctly.  Note the indentation lining up with the `C` in `Create`.

File: cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst [hunk 7]
```diff
@@ -6581,8 +6616,8 @@ One can also change what compilers are written into the generated
 ``<Project>Config.cmake`` and ``<Package>Config.cmake`` files for the build
 and the install trees.  By default, the compilers pointed to in these
 ``Config.cmake`` files will be ``CMAKE_<LANG>_COMPILER`` where ``<LANG>`` =
-``CXX``, ``C``, and ``Fortran``.  But if can change this by setting any of the
-following::
+``CXX``, ``C``, and ``Fortran``.  But if one can change this by setting any of
```
Comment (jmgate) [inline cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst:6619]:
```suggestion
``CXX``, ``C``, and ``Fortran``, but one can change this by setting any of
```

File: cmake/tribits/python_utils/gitdist-setup.sh [hunk 8]
```diff
@@ -40,7 +40,7 @@ SCRIPT_DIR=`echo $BASH_SOURCE | sed "s/\(.*\)\/.*\.sh/\1/g"`
 
 existing_gitdist=`which gitdist 2> /dev/null`
 if [[ "${existing_gitdist}" == "" ]] ; then
-  echo "Setting alias gitdist=${SCRIPT_DIR}/gitdist"
+  #echo "Setting alias gitdist=${SCRIPT_DIR}/gitdist"
```
Comment (jmgate) [inline cmake/tribits/python_utils/gitdist-setup.sh:43]: Remove?

File: cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst (hunk 6 above)
Comment (bartlettroscoe) [reply cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst]: @jmgate, this is addressed in commit https://github.com/TriBITSPub/TriBITS/commit/38cc5ab92224df62d9bce8617249ff1ba1edd3fb and you can see the updated formatting at:

* https://tribits.org/doc/TribitsDevelopersGuide.html#how-to-submit-testing-results-to-a-custom-cdash-group

Thanks for noting this.

File: cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst (hunk 7 above)
Comment (bartlettroscoe) [reply cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst]: Addressed in commit https://github.com/TriBITSPub/TriBITS/commit/c7cc94989ba586612113da5e266ec1201ac63191.

File: cmake/tribits/python_utils/gitdist-setup.sh (hunk 8 above)
Comment (bartlettroscoe) [reply cmake/tribits/python_utils/gitdist-setup.sh]: Addressed in commit https://github.com/TriBITSPub/TriBITS/commit/701f5a1cd0266ad853d45b286258ae7f6c724157

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake (hunk 1 above)
Comment (bartlettroscoe) [reply cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake:1014]: @jmgate, the term "model" is the correct term.  See:

* https://cmake.org/cmake/help/latest/command/ctest_start.html

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake (hunk 2 above)
Comment (bartlettroscoe) [reply cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake]: Addressed in commit https://github.com/bartlettroscoe/TriBITS/commit/c38b89ca8c836b4f9d9ded010b9825aa959b39e0

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake (hunk 3 above)
Comment (bartlettroscoe) [reply cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake]: Addressed in commit https://github.com/bartlettroscoe/TriBITS/commit/c38b89ca8c836b4f9d9ded010b9825aa959b39e0

File: cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake (hunk 4 above)
Comment (bartlettroscoe) [reply cmake/tribits/ctest_driver/TribitsCTestDriverCore.cmake]: Addressed in commit https://github.com/bartlettroscoe/TriBITS/commit/c38b89ca8c836b4f9d9ded010b9825aa959b39e0

**Comment by bartlettroscoe** [comment]
@jmgate, thanks for your review.  I have addressed all of your suggestions in TriBITS commits and updated the TriBITS snapshot in this Trilinos branch.

Also note that this has fixed the problem with truncated test output.  See https://github.com/TriBITSPub/TriBITS/issues/326#issuecomment-685761971.

Can you please approve this PR so it can be merged?

**Pull request merged**